src/modules/InstURL.rb

```diff
@@ -142,232 +83,68 @@ def RewriteCDUrl(url)
-        if proxyuser != nil && proxyuser != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxyuser="),
-            proxyuser
-          )
-          option_separator = "&"
-        end
-        proxypassword = Linuxrc.InstallInf("ProxyPassword")
-        if proxypassword != nil && proxypassword != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "proxypassword="),
-            proxypassword
-          )
-          option_separator = "&"
-        end
-        workgroup = Linuxrc.InstallInf("WorkDomain")
-        if workgroup != nil && workgroup != ""
-          url = Ops.add(
-            Ops.add(Ops.add(url, option_separator), "workgroup="),
-            workgroup
-          )
-          option_separator = "&"
-        end
-
-        if instmode == "https"
-          Builtins.y2milestone("HTTPS instmode detected")
+      log.info "Using install URL: #{URL.HidePassword(@installInf2Url)}"
+      @installInf2Url
+    end
 
-          if !SSLVerificationEnabled()
-            Builtins.y2security(
-              "Disabling certificate check for the installation repository"
-            )
+    # Schemes considered local for installInf2Url
+    LOCAL_SCHEMES = ["cd", "dvd", "hd"]
 
-            # libzypp uses ssl_verify=no option
-            url = Ops.add(Ops.add(url, option_separator), "ssl_verify=no")
-            option_separator = "&"
-          end
-        end
-      end # is_network
+    # Determines whether the installation URL is remote or not
+    #
+    # @return [Boolean] true if it's remote; false otherwise.
+    # @see installInf2Url
+    def is_network
+      return @is_network unless @is_network.nil?
+      scheme = URL.Parse(installInf2Url("")).fetch("scheme")
+      @is_network = !LOCAL_SCHEMES.include?(scheme.downcase)
+    end
 
-      if options != ""
-        url = Ops.add(Ops.add(url, option_separator), options)
-        option_separator = "&"
-        Builtins.y2milestone("options %1", options)
-      end
+  private
 
-      url = Builtins.sformat("iso:/?iso=%1&url=%2", isoimg, url) if isoimg != ""
+    # Helper method to add extra_dir to a given URL
+    #
+    # @param url       [String] URL
+    # @param extra_dir [String] Path to add
+    # @return [String] URL with the added path
+    def add_extra_dir_to_url(url, extra_dir)
+      parts = URL.Parse(url)
+      parts["path"] = File.join(parts["path"], extra_dir)
+      URL.Build(parts)
+    end
 
-      Builtins.y2debug("URL %1", URL.HidePassword(url))
-      url
+    # Helper method to add ssl_verify parameter if needed
+    #
+    # Only applicable if scheme is 'https'.
+    #
+    # @param url       [String] URL
+    # @return [String] URL (with ssl_verify set to 'no' if needed)
+    def add_ssl_verify_no_to_url(url)
+      parts = URL.Parse(url)
+      return url if parts["scheme"].downcase != "https"
+      log.error "Disabling certificate check for the installation repository"
```

Comment:
I agree. But I found [this](https://github.com/yast/rubocop-yast/blob/0cdddc26bead94730322dbf4d99f8002318082f0/lib/rubocop/yast/builtins/y2log.rb#L15).